src/parser/whamm_parser.rs

```diff
@@ -486,6 +731,41 @@ fn stmt_from_rule(pair: Pair<Rule>, err: &mut ErrorGen) -> Result<Statement, Vec
                 loc: Some(Location::from(&var_id_line_col, &var_id_line_col, None)),
             })
         }
+        //HAS NOT BEEN CHECKED
+        Rule::return_statement => {
+            trace!("Entering return_stmt");
+            let mut pair = pair.into_inner();
+            let expr_rule = pair.next().unwrap();
+
+            return match expr_from_pair(expr_rule) {
+                Err(errors) => {
+                    err.add_errors(errors);
```

Comment:
Okay, this is super interesting...I don't actually know how this happened, but somehow this function has access to `ErrorGen`, but is also returning a `Result` type. Instead, it should put all errors in `ErrorGen` (like you did) and return statements _not_ wrapped in a `Result`.

Could you go change that for this function?